core/api/transport/impl/ws/ws_session.hpp

```diff
@@ -151,12 +153,10 @@ namespace kagome::api {
     Configuration config_;  ///< session configuration
     boost::beast::websocket::stream<boost::asio::ip::tcp::socket &> stream_;
     boost::beast::flat_buffer rbuffer_;  ///< read buffer
-    boost::beast::flat_buffer wbuffer_;  ///< write buffer
 
-    std::mutex cs_;
     std::queue<std::string> pending_responses_;
 
-    std::atomic_bool writing_in_progress_ = false;
+    bool writing_in_progress_ = false;
```

Comment:
Hm, what changed so we don't need thread synchronization here anymore? Or was it never needed?

Reply:
We replaced mutex with asio `post` (sequential execution in queue)